xmtp_mls/src/storage/encrypted_store/mod.rs

```diff
@@ -198,6 +194,41 @@ impl EncryptedMessageStore {
         crypto_utils::rng().fill_bytes(&mut key[..]);
         key
     }
+
+    pub async fn transaction_async<T, F, E, Fut>(&self, fun: F) -> Result<T, E>
+    where
+        F: FnOnce(XmtpOpenMlsProvider) -> Fut,
```

Comment:
We can make this `FnOnce(&XmtpOpenMlsProvider)` by using a `BoxFuture` and be consistent with `fn transaction` but it seems unnecessary to box something if we don't have to.

There are some fussy alternatives using traits and stuff but until rust has real `async_closures` the lifetime issue of making the future live as long as the closure arguments w/o having to box the future is a limitation of rust